File: src/main/scala/io/tarantool/spark/connector/rdd/api/java/TarantoolJavaRDD.scala
package io.tarantool.spark.connector.rdd.api.java

import io.tarantool.spark.connector.rdd.TarantoolRDD
import org.apache.spark.api.java.JavaRDD

import scala.reflect.ClassTag

class TarantoolJavaRDD[R](override val rdd: TarantoolRDD[R])(
  override implicit val classTag: ClassTag[R]
) extends JavaRDD[R](rdd) {}

File: src/main/java/io/tarantool/spark/connector/rdd/TarantoolJavaRDD.java
package io.tarantool.spark.connector.rdd;

import org.apache.spark.api.java.JavaRDD;
import scala.reflect.ClassTag;

import static io.tarantool.spark.connector.util.ScalaToJavaHelper.getClassTag;

/**
 * Bridge from Scala {@link TarantoolReadRDD} to Java.
 *
 * Instances of this class may be instantiated using methods of {@link SparkContextJavaFunctions}.
 *
 * @param <R> target POJO type
 */
public class TarantoolJavaRDD<R> extends JavaRDD<R> {

    public TarantoolJavaRDD(TarantoolReadRDD<R> rdd, Class<R> clazz) {
        super(rdd, getClassTag(clazz));
    }

    public TarantoolJavaRDD(TarantoolReadRDD<R> rdd, ClassTag<R> classTag) {
        super(rdd, classTag);
    }

    @Override
    public TarantoolReadRDD<R> rdd() {
        return (TarantoolReadRDD<R>) super.rdd();
    }
}
